src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```
@@ -78,13 +78,33 @@ export const defaultPrepareQueryString = (query: Query): string => {
       );
   }
 };
 
+/**
+ * Checks if query execution should be skipped for the given query.
+ * This provides a centralized place to add language-specific skip conditions.
+ */
+export const shouldSkipQueryExecution = (query: Query): boolean => {
+  switch (query.language) {
+    case 'PROMQL':
+      const queryValue = query.query;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)